src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -348,34 +357,39 @@ private SyntaxNode GetNodeContainingTargetNode(SyntaxNode matchingNode)
                     isReducedExtensionMethod = true;
                 }
 
-                var newArguments = PermuteArgumentList(document, declarationSymbol, invocation.ArgumentList.Arguments, signaturePermutation, isReducedExtensionMethod);
-                return invocation.WithArgumentList(invocation.ArgumentList.WithArguments(newArguments).WithAdditionalAnnotations(changeSignatureFormattingAnnotation));
+                var argumentList = invocation.ArgumentList;
+                var newArguments = PermuteArgumentList(declarationSymbol, argumentList.Arguments, argumentList.OpenParenToken, signaturePermutation, isReducedExtensionMethod);
+                return invocation.WithArgumentList(argumentList.WithArguments(newArguments).WithOpenParenToken(argumentList.OpenParenToken.WithTrailingTrivia()).WithAdditionalAnnotations(changeSignatureFormattingAnnotation));
             }
 
             if (updatedNode.IsKind(SyntaxKind.ObjectCreationExpression, out ObjectCreationExpressionSyntax objCreation))
             {
-                var newArguments = PermuteArgumentList(document, declarationSymbol, objCreation.ArgumentList.Arguments, signaturePermutation);
-                return objCreation.WithArgumentList(objCreation.ArgumentList.WithArguments(newArguments).WithAdditionalAnnotations(changeSignatureFormattingAnnotation));
+                var argumentList = objCreation.ArgumentList;
+                var newArguments = PermuteArgumentList(declarationSymbol, argumentList.Arguments, argumentList.OpenParenToken, signaturePermutation);
+                return objCreation.WithArgumentList(argumentList.WithArguments(newArguments).WithOpenParenToken(argumentList.OpenParenToken.WithTrailingTrivia()).WithAdditionalAnnotations(changeSignatureFormattingAnnotation));
             }
 
             if (updatedNode.IsKind(SyntaxKind.ThisConstructorInitializer) ||
                 updatedNode.IsKind(SyntaxKind.BaseConstructorInitializer))
             {
                 var constructorInit = (ConstructorInitializerSyntax)updatedNode;
-                var newArguments = PermuteArgumentList(document, declarationSymbol, constructorInit.ArgumentList.Arguments, signaturePermutation);
-                return constructorInit.WithArgumentList(constructorInit.ArgumentList.WithArguments(newArguments).WithAdditionalAnnotations(changeSignatureFormattingAnnotation));
+                var argumentList = constructorInit.ArgumentList;
+                var newArguments = PermuteArgumentList(declarationSymbol, argumentList.Arguments, argumentList.OpenParenToken, signaturePermutation);
+                return constructorInit.WithArgumentList(argumentList.WithArguments(newArguments).WithOpenParenToken(argumentList.OpenParenToken.WithTrailingTrivia()).WithAdditionalAnnotations(changeSignatureFormattingAnnotation));
```

Comment:
1. We have .WithoutTrailingTrivia f this purpose.  It would be a lot cleaner IMO.
2. There is a lot of duplication of all these sections.  Can we extract out a common helper and pass in the necessary pieces of data into it?

Reply:
I don't think that there's a WithoutTrailingTrivia method that applies to SyntaxToken (or at least I couldn't find one). I could create one if you think that would be the best course of action.

Reply:
Yup.  That would be best.  There's no reason not to have one, and not to maintain symmetry with the SyntaxNode equivalents.